packages/mobile/src/components/collection-list/CollectionCard.tsx

```diff
@@ -1,138 +1,145 @@
-import { createElement, useCallback, useMemo } from 'react'
+import { useCallback } from 'react'
 
-import { SquareSizes } from '@audius/common/models'
-import type { Collection, ID } from '@audius/common/models'
+import type { ID } from '@audius/common/models'
 import {
-  cacheCollectionsSelectors,
-  reachabilitySelectors
+  DogEarType,
+  SquareSizes,
+  isContentUSDCPurchaseGated
+} from '@audius/common/models'
+import {
+  accountSelectors,
+  cacheCollectionsSelectors
 } from '@audius/common/store'
-import type { CommonState } from '@audius/common/store'
-import type { StyleProp, ViewStyle } from 'react-native'
+import { formatCount } from '@audius/common/utils'
+import type { GestureResponderEvent } from 'react-native'
 import { useSelector } from 'react-redux'
 
-import type { ImageProps } from '@audius/harmony-native'
-import { Card } from 'app/components/card'
-import { CollectionImage } from 'app/components/image/CollectionImage'
+import {
+  Divider,
+  Flex,
+  IconHeart,
+  IconRepost,
+  Paper,
+  Text
+} from '@audius/harmony-native'
+import { UserLink } from 'app/components/user-link'
 import { useNavigation } from 'app/hooks/useNavigation'
-import { useOfflineTracksStatus } from 'app/hooks/useOfflineTrackStatus'
-import { OfflineDownloadStatus } from 'app/store/offline-downloads/slice'
-import { formatCount } from 'app/utils/format'
+
+import { DogEar, LockedStatusBadge } from '../core'
+import { CollectionImage } from '../image/CollectionImage2'
 
 const { getCollection } = cacheCollectionsSelectors
-const { getIsReachable } = reachabilitySelectors
+const { getUserId } = accountSelectors
 
-const formatPlaylistCardSecondaryText = (saves: number, tracks: number) => {
-  const savesText = saves === 1 ? 'Favorite' : 'Favorites'
-  const tracksText = tracks === 1 ? 'Track' : 'Tracks'
-  return `${formatCount(saves)} ${savesText} • ${tracks} ${tracksText}`
+const messages = {
+  repost: 'Reposts',
+  favorites: 'Favorites',
+  hidden: 'Hidden'
 }
 
-type FullCollectionCardProps = {
-  collection: Collection
-  style?: StyleProp<ViewStyle>
-  onPress?: (id: ID) => void
-  /** Override for what number to show as the # of tracks. Optional. */
-  numTracks?: number
+type CollectionCardProps = {
+  id: ID
+  onPress?: (e: GestureResponderEvent) => void
 }
 
-type CollectionCardWithIdProps = {
-  collectionId: ID
-  style?: StyleProp<ViewStyle>
-  onPress?: (id: ID) => void
-}
+export const CollectionCard = (props: CollectionCardProps) => {
+  const { id, onPress } = props
 
-type CollectionCardProps = FullCollectionCardProps | CollectionCardWithIdProps
+  const collection = useSelector((state) => getCollection(state, { id }))
```

Comment:
could use audius-query here

Reply:
yeah i want to wait until audius-query can do batching requests, but so down!!